pyani_plus/private_cli.py

```diff
@@ -426,6 +426,80 @@ def fragment_fasta(
     return 0
 
 
+@app.command()
+def log_anim(
+    database: Annotated[str, typer.Option(help="Path to pyANI-plus SQLite3 database")],
+    # These are for the comparison table
+    query_fasta: Annotated[Path, typer.Option(help="Path to query FASTA file")],
+    subject_fasta: Annotated[Path, typer.Option(help="Path to subject FASTA file")],
+    deltafilter: Annotated[Path, typer.Option(help="Path to deltafilter output file")],
+    # Don't use any of fragsize, maxmatch, kmersize, minmatch (configuration table entries)
+) -> int:
+    """Log a single pyANI-plus ANIm pairwise comparison (with nucmer) to the database."""
+    # Assuming this will match as expect this script to be called right
+    # after the computation has finished (on the same machine)
+    nucmer_tool = tools.get_nucmer()
+
+    r_aligned_bases, q_aligned_bases, identity, sim_errors = method_anim.parse_delta(
+        deltafilter
+    )
```

Comment:
As per legacy pyANI v0.3, should we use the query or reference aligned bases as the alignment length for query-vs-subject?

Reply:
We had a conversation about this in https://github.com/widdowquinn/pyani/issues/151! Since, the behaviour of nucmer is not symmetrical (eg. values might _slightly_ differ for A:B and B:A companions), we were left with different choices:
1. Calculation of a summary (max, min, mean) between A:B and B:A.
2. Reporting for the query, such as reporting for genome B in A:B comparisons and genome A in B:A comparisons.
3. Reporting for the reference, for example, reporting for genome A in A:B comparisons and genome B in B:A comparisons.

We agreed that there is no right answer, and we have decided to report values for the query in this case.

Reply:
The old code did this:

```python
        (
            query_tot_length,
            subject_tot_length,
            weighted_identity,
            tot_sim_error,
        ) = parse_delta(deltafile)
```

And logged ``query_tot_length`` as the query-vs-subject alignment length, _and_ ``subject_tot_length`` as the subject-vs-query alignment length.

Does the new ``parse_delta`` return query or subject length first?

Reply:
Quoting old pyANI v0.3,

```
def parse_delta(filename: Path) -> Tuple[int, int, float, int]:
    """Return (reference alignment length, query alignment length, average identity, similarity erors)
    ...
    return (raln_length, qaln_length, avrg_ID, sim_error)
```

Yet when used,

```python
        (
            query_tot_length,
            subject_tot_length,
            weighted_identity,
            tot_sim_error,
        ) = parse_delta(deltafile)
```

It does look like query and reference (aka subject) got flipped there?

Reply:
Quoting pyANI-plus,

```
def parse_delta(filename: Path) -> tuple[int, int, float, int]:
    """Return (reference alignment length, query alignment length, average identity, similarity errors).
    ...
    return (
        get_aligned_bases_count(regions_ref),
        get_aligned_bases_count(regions_qry),
        avrg_identity,
        sim_error,
    )
```

So this is reference then query

Reply:
Well... I'm getting a little bit confused here. I refer to them as reference_vs_query, because that's what nucmer refers to them as. In new `pyANI`, the code reported the values for the query. 

So far, the tests that were carried out all match between old and new pyani, so perhaps there were swapped  somewhere?

Reply:
I _believe_ the tests on this PR confirm we're matching legacy pyANI output, but all the same, this line in old pyANI appears to swap the variable names:

```python
        (
            query_tot_length,
            subject_tot_length,
            weighted_identity,
            tot_sim_error,
        ) = parse_delta(deltafile)
```

We could also have different interpretations of subject vs query and the matrix output (transposed or not)? On the brightside, the expected ANIm alignment length matrix is  asymmetric for the three viruses.

Reply:
😱

Well, I guess this matches the tested behaviour in pyANI v0.3 and that's OK?

Reply:
At this point, I’m not sure... Since I started working on pyANI in December, I haven’t touched this part of the code. The only thing I fixed was the issue in parse_delta, where duplicated regions were being counted twice.

When @widdowquinn and I agreed to report values for the query, I assumed he meant the term "query" as used in the nucmer documentation. Honestly, I’m not sure why the old pyANI refers to them as query vs subject. Personally, I’d lean toward using reference vs query—it’s clearer and aligns better with `nucmer` documentation. Sticking to consistent terminology would make things easier for end-users and anyone else working on this project in the future. I can be wrong tho 😭

Reply:
> Well, I guess this matches the tested behaviour in pyANI v0.3 and that's OK?

The calculations in legacy `pyANI` and `pyANI-plus`, should be correct as I remember calculating the values by hand from `.filter` files, when trying to fix the genome coverage issue. The question is what value we would want to report... `query`, `subject`, or `reference`?

Reply:
I think the best way to go about this is to report `query` (eg. `query` as in `nucmer` documentation), as this is the sequence that is being compared.

Reply:
NCBI BLAST uses the terms query vs subject, and so currently do our column names in the table. I'm used to having subject interchangeable with reference for pairwise alignments.

Reply:
My apologies for my clumsy wording here. I absolutely agree with you that using subject interchangeably with reference for pairwise alignment is okay. I just don't think we should refer to it as query vs. subject/reference (when constructing a command in `Snakemake`) in the context of aniM comparisons, since the comparisons are reference/subject vs. query. This could cause issues, like the one we encountered with the old `pyANI`, if that makes sense.

Reply:
I looked at the `parse_delta` function in `pyANI-plus`, and it reportes the correct values for what is consiedered a `query` and `reference` in `nucmer`. 

If you happy with work done on this branch, feel free to merge this into main.

Reply:
Thank you